Update event-driven-content.md (#4581)
MicrosoftDocs/architecture-center: docs/guide/architecture-styles/event-driven-content.md

> Event-driven architecture is a popular style for building applications that are fault-tolerant, modular, highly scalable, and capable of evolving quickly. It consists of **event producers** that generate a stream of events, **event consumers** that listen for these events, and **event channels**—typically message queues or pub/sub topics—that transfer events between producers and consumers

Reviewer: ```suggestion
An event-driven architecture consists of **event producers** that generate a stream of events, **event consumers** that listen for these events, and **event channels** that transfer events from producers to consumers.
```

> - Producers and consumers are decoupled.

Author: I'm thinking to mention the evolutionary feature of EDA in here. For example,

> No point-to-point integrations are required, making it easy to add new consumers to the system. This allows for easier evolution of the system.

What's your advice?

Author: Being "evolutionary" is encouraged in another page, so I would suggest mentioning it here. https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/design-for-evolution

Reviewer: You can add that, but please avoid terms like "easy" - that's not for you to decide and is so non specific, it sounds like marketing junk.

Just state what & how it supports workload evolution over time. Let's do that in a different PR. I need to move forward today and don't want this one blocked

- No point-to-point integrations. It's easy to add new consumers to the system.
- Consumers can respond to events immediately as they arrive.

> - Highly scalable, elastic and distributed.

Reviewer: ```suggestion
- Highly scalable, elastic, and distributed.
```

- While a request is only visible to the request-handling component, events are often visible to multiple components in a workload, even if those components don't or aren't meant to consume them. Operating with an "assume breach" mindset, be mindful of what information you include in events to prevent unintended information exposure.
- Many applications use event-driven architecture as their primary architecture; however, this approach can be combined with other architectural styles, resulting in hybrid architectures. Common combinations include [microservices](./microservices.yml) and [pipes and filters](../../patterns/pipes-and-filters.yml). Integrating event-driven architecture enhances system performance by eliminating bottlenecks and providing [back pressure](https://wikipedia.org/wiki/Back_pressure) during high request volumes.

> - Any [specific domain](../../microservices/model/domain-analysis.md) spans multiple event producers, consumers, and ingestion channels. Therefore, changes to a particular domain typically impact many components.

Reviewer: ```suggestion
- [Specific domains](../../microservices/model/domain-analysis.md) often span multiple event producers, consumers, or event channels. Changes to a particular domain might impact many components.
```
